pkg/event/event.go

```diff
@@ -9,7 +9,8 @@ import (
 
 // Event represents the canonical representation of a CloudEvent.
 type Event struct {
-	Context     EventContext
+	Context EventContext
+	// Data can be a string or a []byte
```

Comment:
Data can also be a struct, map, or array. Really, it can be anything that we can call json.Marshal on

Reply:
Ah that was not clear to me, how should i implement a deep copy for them?

Reply:
you will hate this... but I think the surest way in go is to round trip it though json.Mashal/Unmarshal, but that only works if the objects are intended as json.